libs/remix-core-plugin/src/lib/compiler-artefacts.ts

```diff
@@ -72,6 +72,16 @@ export class CompilerArtefacts extends Plugin {
     return contractsData
   }
 
+  getArtefactsByContractName (contractName) {
+    const contractsDataByFilename = this.getAllContractDatas()
```

Comment:
it's maybe better to start checking the very last compilation result `getLastCompilationResult`.

Reply:
current logic covers all compilations

Reply:
yes that's right, but if multiple contracts with the same name were compiled, by using `getAllContractDatas` you are unsure to get the last compiled one... which is the one that it's likely users want to interact with.
that might not happen often but still...